### DISCUSSION TITLE
The KSP of my framework conflicts with the KSP of micronaut

### DISCUSSION THREAD

**Discussion by babyfish-ct** [micronaut-projects/micronaut-core#10388] [Q&A]
Help please.

I created a framework called [jimmer](https://github.com/babyfish-ct/jimmer) which is used by some users via spring or quarkus.

A few days ago, I received the [issue-378](https://github.com/babyfish-ct/jimmer/issues/378) submitted by a user indicating that jimmer's ksp conflicts with micronaut's ksp. I reproduced the problem following the user's description. The project is in the attachment, Directly download it can reproduce the error very fast.

I'm unable to find the cause of the problem, so I don't know how to respond to the user. I'm requesting help. If I want jimmer's ksp to be compatible with micronaut ksp, what issues should I pay attention to?

---

I found the last micronaut-related error message is:

```
io.micronaut.kotlin.processing.ExtensionsKt.getBinaryName(extensions.kt:50)
```

It's probably because my ksp generates some Kotlin extension functions.

Here is the attachment, [demo.zip](https://github.com/micronaut-projects/micronaut-core/files/13988919/demo.zip)

**Comment by dstepanov** [comment]
I think it might be fixed in the non-released Core 4.3. The release should be soon.

**Comment by babyfish-ct** [comment]
Cool
